### ISSUE TITLE
Reduce noise

### ISSUE THREAD

**Issue by roganov** [ambv/black#72]
IMO `already well formatted, good job` lines are pretty useless, especially on big projects. Currently it's not possible to silence them without some bash magic since all output is printed to stderr. Would it make sense to provide an option to not print these lines?

**Comment by ambv** [comment]
tl;dr - I don't plan for more fine-grained control over output than what's currently available.

When multiple files (or a directory) are given on the command line, the user wants to know which files matched. This is why Black outputs a line for every file. Whether the file was changed or not is secondary.

If this information is useless for you, you can redirect it to a file. It's always produced on stderr so the "Bash magic" is as simple as `black . 2>/tmp/black_last.log`. You could even redirect to `/dev/null` but keeping it in a file instead is helpful in the case when it returns with a non-zero exit code, indicating an internal error. Most integrations (like PyCharm's File Watchers) will do this automatically for you.

If you would like to only know about files that changed, you can `grep -v "good job" /tmp/black_last.log`. I don't think every small tweak like this should be implemented in every tool.

**Comment by roganov** [comment]
I agree with you that more configuration options is not ideal, but maybe these "good job" lines are not needed at all?
> When multiple files (or a directory) are given on the command line, the user wants to know which files matched.

I very much doubt that people really check if black matched every file in the provided path.

Other tools such as isort and flake8 don't print anything for 'good' files.

**Comment by ambv** [comment]
I disagree. When you try out Black for the first time and it won't do anything with your file, you'll be wondering if it even works. I think an encouragement saying "hey, I looked at your file and wouldn't change a thing!" is just good user experience.

If you only want to see which files changed, `git status` or `find . -mtime -1m` will tell you.

Flake8 is different because it's goal is to produce warnings. No warnings = good. It doesn't modify files. I don't know about isort's standard output, I don't use it.

Sorry that I'm protecting the status quo so stubbornly here but I'm pretty sure the minute I tweaked it, somebody else would open an issue asking for the opposite thing. So I'm going to leave this as is.

**Comment by roganov** [comment]
I very much respect your decision, but wanted to say that user experience won't be hurt since black prints summary at the very end.
```
All done! ✨ 🍰 ✨
1 file reformatted, 788 files left unchanged.
```
